File: AndroidCourse/Java/Lesson_1/app/src/main/java/com/example/lesson_1/Activity1.java
package com.example.lesson_1;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.TreeSet;

public class Activity1 extends AppCompatActivity {
    private TreeSet<String> studentList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_1);

        studentList = new TreeSet<>();

        Button buttonSaveInArray = findViewById(R.id.buttonSaveInArrayAct1);
        EditText editTextInputStudent = findViewById(R.id.editTextInputStudentAct1);
        buttonSaveInArray.setOnClickListener(view -> {
            if (!editTextInputStudent.getText().toString().trim().isEmpty()) {
                String name = editTextInputStudent.getText().toString();
                studentList.add(name);
                editTextInputStudent.setText("");

                Toast.makeText(Activity1.this, String.format(getString(R.string.success_save_student_text_act1), name), Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(Activity1.this, R.string.empty_editTextInputStudent_text_act1, Toast.LENGTH_SHORT).show();
            }
        });

        Button buttonOutput = findViewById(R.id.buttonOutputAct1);
        TextView textViewOutput = findViewById(R.id.textViewOutputAct1);
        buttonOutput.setOnClickListener(view -> {
            textViewOutput.setText("");

            for (String student : studentList) {
                String name = textViewOutput.getText() + student + "\n";
                textViewOutput.setText(name);
            }

            hideKeyboardFrom(Activity1.this, buttonOutput);

            Toast.makeText(Activity1.this, String.format(getString(R.string.output_message_text_act1), Integer.toString(studentList.size())), Toast.LENGTH_SHORT).show();
        });
    }

    private static void hideKeyboardFrom(Context context, View view) {
        InputMethodManager imm = (InputMethodManager) context.getSystemService(Activity.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: AndroidCourse/Kotlin/Lesson_1/app/src/main/java/com/example/lesson_1/Activity1.kt
package com.example.lesson_1

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_1.*
import java.util.*

class Activity1 : AppCompatActivity() {
    private var studentList: TreeSet<String> = TreeSet()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_1)

        buttonSaveInArrayAct1.setOnClickListener {
            if (editTextInputStudentAct1.text.toString().trim().isNotEmpty()) {
                val name = editTextInputStudentAct1.text.toString()
                studentList.add(name)
                editTextInputStudentAct1.setText("")

                Toast.makeText(
                    this,
                    String.format(getString(R.string.success_save_student_text_act1), name),
                    Toast.LENGTH_SHORT
                ).show()
            } else {
                Toast.makeText(
                    this,
                    R.string.empty_editTextInputStudent_text_act1,
                    Toast.LENGTH_SHORT
                ).show()
            }
        }

        buttonOutputAct1.setOnClickListener {
            textViewOutputAct1.text = ""
            for (student in studentList) {
                val newStudentLine = "${textViewOutputAct1.text}$student\n"
                textViewOutputAct1.text = newStudentLine
            }

            buttonOutputAct1.hideKeyboard()

            Toast.makeText(
                this,
                String.format(getString(R.string.output_message_text_act1), studentList.size),
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private fun View.hideKeyboard() {
        val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(windowToken, 0)
    }
}